openedx/core/djangoapps/catalog/utils.py

```diff
@@ -54,11 +55,18 @@ def get_programs(uuid=None, types=None):  # pylint: disable=redefined-builtin
         )
 
         querystring = {
-            'marketable': 1,
             'exclude_utm': 1,
         }
+
+        # TODO ECOM-7650: Remove this after https://github.com/edx/course-discovery/pull/805 is merged and released.
+        if waffle.switch_is_active('display_retired_programs_on_learner_dashboard'):
+            querystring['status'] = ('active', 'retired',)
```

Comment:
This change opens us up to the possibility of programs without marketing slugs appearing in the LMS, but I don't think that's a problem. There was a period, pre-detail page, when that URL was necessary, but we don't use it anymore.